validphys2/src/validphys/photon/alpha.py

```diff
@@ -0,0 +1,241 @@
+        elif self.theory["ModEv"] == "EXA":
+            self.alphaem_fixed_flavor = self.alphaem_fixed_flavor_exa
+            self.thresh, self.alphaem_thresh = self.compute_alphaem_at_thresholds()
+
+            xmin = XGRID[0]
+            qmin = xmin * theory["MP"] / np.sqrt(1 - xmin)
+            # use a lot of interpolation points since it is a long path 1e-9 -> 1e4
+            self.q = np.geomspace(qmin, np.sqrt(q2max), 500, endpoint=True)
+
+            # add threshold points in the q list since alpha is not smooth there
+            self.q = np.append(
+                self.q, [ME, MMU, MQL, self.thresh_c, MTAU, self.thresh_b, self.qref, self.thresh_t]
+            )
+            self.q = self.q[np.isfinite(self.q)]
+            self.q.sort()
+
+            self.alpha_vec = np.array([self.alpha_em(q_) for q_ in self.q])
+            self.alpha_em = self.interpolate_alphaem
+        else:
+            raise ValueError(f"Evolution mode not recognized: {self.theory['ModEv']}")
+
+    def interpolate_alphaem(self, q):
+        r"""
+        Interpolate precomputed values of alpha_em.
+
+        Parameters
+        ----------
+        q: float
+            value in which alpha_em is computed
+
+        Returns
+        -------
+        alpha_em: float
+            electromagnetic coupling
+        """
+        return np.interp(q, self.q, self.alpha_vec)
+
+    def alpha_em(self, q):
+        r"""
+        Compute the value of the running alphaem.
+
+        Parameters
+        ----------
+        q: float
+            value in which alphaem is computed
+
+        Returns
+        -------
+        alpha_em: numpy.ndarray
+            electromagnetic coupling
+        """
+        nf, nl = self.find_region(q)
+        return self.alphaem_fixed_flavor(
+            q, self.alphaem_thresh[(nf, nl)], self.thresh[(nf, nl)], nf, nl
+        )
+
+    def alphaem_fixed_flavor_trn(self, q, alphaem_ref, qref, nf, nl):
+        """
+        Compute the running alphaem for nf fixed at NLO, using truncated method.
+        In this case the RGE for alpha_em is solved decoupling it from the RGE for alpha_s
+        (so the mixed terms are removed). alpha_s will just be unused.
+
+        Parameters
+        ----------
+        q : float
+            target scale
+        alph_aem_ref : float
+            reference value of alpha_em
+        qref: float
+            reference scale
+        nf: int
+            number of flavors
+
+        Returns
+        -------
+        alpha_em at NLO : float
+            target value of a
+        """
+        if (nf, nl) == (0, 0):
+            return alphaem_ref
+        lmu = 2 * np.log(q / qref)
+        den = 1.0 + self.betas_qed[(nf, nl)][0] * alphaem_ref * lmu
+        alpha_LO = alphaem_ref / den
+        alpha_NLO = alpha_LO * (
+            1 - self.betas_qed[(nf, nl)][1] / self.betas_qed[(nf, nl)][0] * alpha_LO * np.log(den)
+        )
```

Comment:
I think `self.betas_qed[(nf, nl)][1]` would not exist if QED is at LO. I think there should be some "if theory qed = 1" to return alpha_LO